x/tier/keeper/begin_blocker.go

```diff
@@ -1,15 +1,14 @@
 package keeper
 
 import (
-	"time"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 func (k Keeper) BeginBlocker(ctx sdk.Context) {
-	year, month, day := ctx.BlockTime().Date()
-	dateToday := time.Date(year, month, day, 0, 0, 0, 0, ctx.BlockTime().Location())
-	deleteDate := dateToday.AddDate(0, 0, -8)
+	deleteDate := k.GetDateAfterDaysFromContext(ctx, -8)
 	// Remove last 100 values at each block
-	k.RemovePortfolioLast(ctx, deleteDate.String(), 100)
+	k.RemovePortfolioLast(ctx, deleteDate, 100)
+	// migration does not delete all older entries as we don't know whats last date is there, deleting from past 105 days
```

Comment:
I think we can put this logic in migration. will that not work ?

Reply:
Older entries might take a lot of time to finish deleting leading to workflow failure. So deleting in abci.